scaleway/resource_redis_cluster.go

```diff
@@ -0,0 +1,192 @@
+package scaleway
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	redis "github.com/scaleway/scaleway-sdk-go/api/redis/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayRedisCluster() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayRedisClusterCreate,
+		ReadContext:   resourceScalewayRedisClusterRead,
+		UpdateContext: resourceScalewayRedisClusterUpdate,
+		DeleteContext: resourceScalewayRedisClusterDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultRedisClusterTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the redis cluster",
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Redis version of the cluster",
+				ForceNew:    true,
+			},
+			"node_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Type of node to use for the cluster",
+				ForceNew:    true,
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the user created when the cluster is created",
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Sensitive:   true,
+				Required:    true,
+				Description: "Password of the user",
+			},
+			// Common
+			"zone":       zoneSchema(),
+			"project_id": projectIDSchema(),
+		},
+	}
+}
+
+func resourceScalewayRedisClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	redisAPI, zone, err := redisAPIWithZone(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	createReq := &redis.CreateClusterRequest{
+		Zone:            zone,
+		ProjectID:       d.Get("project_id").(string),
+		Name:            expandOrGenerateString(d.Get("name"), "redis"),
+		Version:         d.Get("version").(string),
+		Tags:            nil,
```

Comment:
If you don't support them, do not specify them. Go will provide the zero values. For Cluster size, acl rules, ...